cpp/tests/transform/integration/unary-operation-integration-test.cu

```diff
@@ -0,0 +1,220 @@
+/*
```

Comment:
The location/naming of this file is a bit non-standard.

Can it just be `tests/transform/unary_tests.cu`?